mailet/api/src/main/java/org/apache/mailet/AttributeUtils.java

```diff
@@ -0,0 +1,52 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.mailet;
+
+import java.util.Optional;
+
+/** 
+ * Attribute
+ * 
+ * @since Mailet API v3.2
+ */
+public class AttributeUtils {
+    public static <T> Optional<T> getValueAndCastFromMail(Mail mail, AttributeName name, Class<T> type) {
+        return getAttributeValueFromMail(mail, name)
+                .flatMap(value -> tryToCast(type, value));
```

Comment:
flatMapping the failed cast looks surprising to me.
Either we want that util class to be part of the API and we have to document its behavior or we don't and we probably should avoid surprising `empty` to occur for a type problem

Reply:
Let's make it part of the API and document it